Data/ByteString/Char8.hs

```diff
@@ -968,9 +968,7 @@ lines (BS x l) = go x l
 -- | 'unlines' is an inverse operation to 'lines'.  It joins lines,
 -- after appending a terminating newline to each.
 unlines :: [ByteString] -> ByteString
-unlines [] = empty
-unlines ss = concat (List.intersperse nl ss) `append` nl -- half as much space
-    where nl = singleton '\n'
+unlines = concat . List.concatMap (\x -> [x, singleton '\n'])
```

Comment:
This is better than before, because it avoids copying the full output twice (with and without final `nl`).
Ideally `unlines` should be implemented in a style similar to `intercalate`, avoiding `concat` (which is quite ineffective):
https://github.com/haskell/bytestring/blob/5efd6b504eccf89feba13a6bf5e0e26eed780da9/Data/ByteString.hs#L1209-L1227